### DISCUSSION TITLE
Shouldn't be clearMocks set to true by default?

### DISCUSSION THREAD

**Discussion by adrianbienias** [vitest-dev/vitest#3012] [General]
By design, unit test should be independent of each other.
Without clearing mocks, the state of them is passed to next tests, which is not a behavior that is expected.

I've just spent an hour trying to figure out why mocks announced themselves as called, when I didn't call them in a test suite.
I guess I'm not the only one who faced/will face such unintuitive approach.

What do you guys think?

**Comment by sheremet-va** [comment]
This is how it's been since the feature was introduced in Jest. While I agree that tests should be isolated by default, I don't think Vitest can force that.
